dll/win32/shell32/systray.cpp

```diff
@@ -51,14 +45,17 @@ BOOL WINAPI Shell_NotifyIconA(DWORD dwMessage, PNOTIFYICONDATAA pnid)
     /* Validate the structure size and the flags */
     cbSize = pnid->cbSize;
     dwValidFlags = NIF_MESSAGE | NIF_ICON | NIF_TIP;
-    if (cbSize == sizeof(NOTIFYICONDATAA))
+#if (_WIN32_WINNT >= _WIN32_WINNT_VISTA)
+    if (cbSize == NOTIFYICONDATAA_V4_SIZE_COMPAT)
     {
-        nidW.cbSize = sizeof(nidW);
+        nidW.cbSize = NOTIFYICONDATAA_V4_SIZE_COMPAT;
```

Comment:
Why switch to `NOTIFYICONDATAA`?
Why are `V4_SIZE_COMPAT` defines even needed?

Reply:
> Why switch to NOTIFYICONDATAA?

Fixed.

> Why are V4_SIZE_COMPAT defines even needed?

To express the logic explicitly.

Reply:
> To express the logic explicitly.

From my point of view, adding `V4_SIZE_COMPAT` adds more complexity/confusion than it helps.
Meanwhile, as is, you should use it at old line 118 too.

I think the same about adding `V3_SIZE_COMPAT`.
Some more code "duplication" should avoid new defines, and/or define `V3_SIZE` for the XP-S2003 case as you attempted for a while.

Reply:
> ...V3_SIZE is not available in XP/S2003 PSDK code.

Yes, that applies to PSDK define: see #5730.
Note:
I do think that committing that PR first would ease/clarify things for everyone...

> How can we support ...V3_SIZE in XP/S2003?

Code duplication: see https://github.com/reactos/reactos/pull/5708/commits/ab203f8acdec22ac9791acc54c8a9539eb1439ed for example
and/or
compatibility define:  see https://github.com/reactos/reactos/pull/5708/commits/768bff6ba5196c9d70bad2166ff6b9f9d8ffa241

Reply:
Ok, I will wait for merging #5730.
Until that, this PR will be pending forever.